pkg/admin/data/data_manager.go

```diff
@@ -91,20 +92,50 @@ func (dm *manager) Backup(name BackupName) error {
 
 	dest := dm.GetBackupPath(name)
 
-	if err := copyDataDir(dest); err != nil {
+	if err := copy(config.DataDir, dest); err != nil {
 		return err
 	}
 
 	klog.InfoS("Backup finished", "backup", dest, "data", config.DataDir)
 	return nil
 }
 
-func (dm *manager) Restore(n BackupName) error {
-	return fmt.Errorf("Restore not implemented")
+func (dm *manager) Restore(name BackupName) error {
+	klog.InfoS("Restoring the data", "storage", dm.storage, "name", name, "data", config.DataDir)
+
+	if name == "" {
+		return &EmptyArgErr{"name"}
+	}
+
+	if exists, err := dm.BackupExists(name); err != nil {
+		return fmt.Errorf("checking if backup %s exists failed: %w", name, err)
+	} else if !exists {
+		klog.ErrorS(nil, "Backup to restore does not exist", "name", name)
+		return fmt.Errorf("backup %s does not exist", name)
+	}
+
+	tmp := fmt.Sprintf("%s.%s", config.DataDir, rand.String(8))
+	klog.InfoS("Temporarily renaming data dir", "data", config.DataDir, "renamedTo", tmp)
```

Comment:
This is asymmetric to `Backup()` (there it's caller's responsibility to ensure the backup dir does not exist).
Should we make caller's responsibility ensure `/var/lib/microshift` does not exist before restoring from a backup?

Reply:
IIUC, `/var/lib/microshift` should never(?) exist before executing the Restore business logic, so it'd be an invariant of Restore(). If that's the case, IMO it should be checked by Restore(), not it's caller. Ditto for Backup().

Reply:
Restore should essentially be

```
mv /var/lib/microshift /var/lib/microshift.save
cp /var/lib/microshift-backups/whatever /var/lib/microshift (using the same flags used when doing the backup)
rm -rf /var/lib/microshift.save
```

If the `cp` fails, then recovery is

```
rm -rf /var/lib/microshift
mv /var/lib/microshift.save /var/lib/microshift
```

I think you want all of that logic managed from 1 function to ensure all of the error handling paths that involve recovery are covered.

Are there files that only exist in `/var/lib/microshift` (and not in the backup) that we need to keep, like history metadata?